src/MetadataManager/MetadataEventManager.ts

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as fs from 'fs';
+import * as vscode from 'vscode';
+
+import {Balloon} from '../Utils/Balloon';
+import {obtainWorkspaceRoot} from '../Utils/Helpers';
+import {Logger} from '../Utils/Logger';
+
+class MetadataEventQueue {
+  private inProgress: boolean = false;
+  private queue: {method: any, input: {[key: string]: any}}[] = [];
+
+  constructor() {
+    this.queue = [];
+  }
+
+  enqueue(method: any, input: {[key: string]: any}): void {
+    this.queue.push({method: method, input: input});
+    this.autoAction();
+  }
+
+  front() {
+    return this.queue[0];
+  }
+  dequeue() {
+    this.queue.shift();
+  }
+
+  clear() {
+    this.inProgress = false;
+    this.queue = [];
+  }
+
+  isEmpty() {
+    return this.queue.length === 0;
+  }
+
+  autoAction() {
+    if (this.inProgress === false) {
+      this.inProgress = true;
+      this.action();
+    }
+  }
+
+  async action() {
+    await this.front().method();
+    this.dequeue();
+
+    if (this.isEmpty()) {
+      this.clear();
+    } else {
+      this.action();
+    }
+  }
+}
+
+class MetadataEventBuffer {
+  private _queue = new MetadataEventQueue();
+  constructor() {}
+  public setEvent(request: any, input: {[key: string]: any}) {
+    this._queue.enqueue(() => {
+      return new Promise(() => {
+        if (Object.keys(input).length === 0) {
+          request();
+        } else {
+          request(input);
+        }
+      });
+    }, input);
+  }
+}
+
+
+
+/* istanbul ignore next */
+export class MetadataEventManager {
+  private fileWatcher = vscode.workspace.createFileSystemWatcher(`**/*`);  // glob pattern
+  private eventBuffer = new MetadataEventBuffer();
+
+    /**
+   * Communicates among events
+   * didCreateUri : 
+   */
+  public static didCreateUri: vscode.Uri|undefined = undefined;
+
+  public static register(context: vscode.ExtensionContext) {
```

Comment:
Please note that Github Actions generates some warnings. You can remove them by this change.
```suggestion
  public static register(_context: vscode.ExtensionContext) {
```

Use 
`npm run lint`
to run this locally.